aws/data_source_aws_mks_cluster.go

```diff
@@ -0,0 +1,60 @@
+package aws
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsMksCluster() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsMksClusterRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"broker_node_group_info": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"encrypt_rest_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enhanced_monitoring": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"kafka_version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"broker_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsMksClusterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kafkaconn
+	arn := d.Get("arn").(string)
+
+	state, err := readMskClusterState(conn, arn)
+	if err != nil {
+		return err
+	}
+	d.SetId(state.arn)
+	d.Set("arn", state.arn)
+	d.Set("status", state.status)
+	d.Set("creation_timestamp", state.creationTimestamp)
+	d.Set("encrypt_rest_arn", state.encryptRestArn)
```

Comment:
You've used `encrypt_rest_key` in the schema here but set this to `encrypt_rest_arn` which is what you're using elsewhere. Personally I find the `encryptRestArn` to be slightly confusing without any mention that it's the KMS key ARN and had to check the source to see that that was the case so I'd switch everything over to `encrypt_rest_key` or some variant on that.

Also it might help to add an acceptance test for this data source even if you are just exercising the same method as the read on the resource just to catch mistakes in the schema like this.